components/application-operator/pkg/application-controller/reconciler.go

```diff
@@ -129,6 +129,8 @@ func (r *applicationReconciler) manageInstallation(application *v1alpha1.Applica
 		}
 
 		return r.installApplication(application)
+	} else if !application.ShouldSkipInstallation() {
+		r.releaseManager.UpgradeApplicationRelease(application)
```

Comment:
This reveals problem. We probably reinstall application whenever any update occurs (reconcile method is called). We should do it only if the update concerns change of override labels for the application.